python/lsst/pex/config/config.py

```diff
@@ -1037,20 +1037,70 @@ def load(self, filename, root="config"):
         See also
         --------
         lsst.pex.config.Config.loadFromStream
+        lsst.pex.config.Config.loadFromString
         lsst.pex.config.Config.save
         lsst.pex.config.Config.saveFromStream
         """
         with open(filename, "r") as f:
             code = compile(f.read(), filename=filename, mode="exec")
-            self.loadFromStream(stream=code, root=root, filename=filename)
+            self.loadFromString(code, root=root, filename=filename)
 
     def loadFromStream(self, stream, root="config", filename=None):
         """Modify this Config in place by executing the Python code in the
         provided stream.
 
         Parameters
         ----------
-        stream : file-like object, `str`, or compiled string
+        stream : file-like object, `str`, `bytes`, or compiled string
+            Stream containing configuration override code.  If this is a
+            code object, it should be compiled with ``mode="exec"``.
+        root : `str`, optional
+            Name of the variable in file that refers to the config being
+            overridden.
+
+            For example, the value of root is ``"config"`` and the file
+            contains::
+
+                config.myField = 5
+
+            Then this config's field ``myField`` is set to ``5``.
+
+            **Deprecated:** For backwards compatibility, older config files
+            that use ``root="root"`` instead of ``root="config"`` will be
+            loaded with a warning printed to `sys.stderr`. This feature will be
+            removed at some point.
+        filename : `str`, optional
+            Name of the configuration file, or `None` if unknown or contained
+            in the stream. Used for error reporting.
+
+        Notes
+        -----
+        For backwards compatibility reasons, this method accepts strings, bytes
+        and code objects as well as file-like objects.  New code should use
+        `loadFromString` instead for most of these types.
+
+        See also
+        --------
+        lsst.pex.config.Config.load
+        lsst.pex.config.Config.loadFromString
+        lsst.pex.config.Config.save
+        lsst.pex.config.Config.saveFromStream
+        """
+        if hasattr(stream, "read"):
+            if filename is None:
+                filename = getattr(stream, "name", "?")
+            code = compile(stream.read(), filename=filename, mode="exec")
+        else:
+            code = stream
+        self.loadFromString(code, root=root, filename=filename)
+
+    def loadFromString(self, code, root="config", filename=None):
```

Comment:
I'm afraid it needs to stay for now because this new method also provides the implementation for the old ones, but I opened DM-32795 to drop support for "root", since I don't want to tackle it here.  It looks like that deprecation just happened well before we had a deprecation policy that said we had to create those kinds of tickets and how long we needed to wait.